src/Appwrite/Network/Validator/DNS.php

```diff
@@ -2,115 +2,93 @@
-        try {
-            $response = $client->query(Message::query(
-                new Question($value, $this->type)
-            ));
-        } catch (\Throwable) {
-            return false;
+        if (\count($this->dnsServers) === 1) {
+            return $this->isValidWithDNSServer($value, $this->dnsServers[0]);
         }
 
-        $typeMatches = array_filter(
-            $response->answers,
-            fn (Record $record) => $record->type === $this->type
-        );
+        $wg = new WaitGroup();
+        $failedValidator = null;
 
-        if (empty($typeMatches)) {
-            if ($this->type === Record::TYPE_CAA) {
-                return $this->validateParentCAA($value);
-            }
+        foreach ($this->dnsServers as $dnsServer) {
+            $wg->add();
 
-            return false;
-        }
+            \go(function () use ($value, $dnsServer, $wg, &$failedValidator) {
+                try {
+                    $validator = new BaseDNS($this->target, $this->type, $dnsServer);
+                    $isValid = $validator->isValid($value);
 
-        foreach ($typeMatches as $record) {
-            if ($this->type === Record::TYPE_CAA) {
-                $valuePart = $this->extractCAAValue($record->rdata);
-                if ($valuePart !== '' && $valuePart === $this->target) {
-                    return true;
+                    if (!$isValid) {
+                        $failedValidator = $validator;
+                    }
+                } finally {
+                    $wg->done();
                 }
-            }
-
-            if ($record->rdata === $this->target) {
-                return true;
-            }
-        }
-
-        return false;
-    }
-
-    private function validateParentCAA(string $domain): bool
-    {
-        try {
-            $domainInfo = new Domain($domain);
-        } catch (\Throwable) {
-            return false;
-        }
-
-        if ($domainInfo->get() === $domainInfo->getApex()) {
-            return true;
+            });
         }
 
-        $parts = explode('.', $domainInfo->get());
-        array_shift($parts);
-        $parent = implode('.', $parts);
+        $wg->wait();
 
-        if ($parent === '') {
+        if (!\is_null($failedValidator)) {
+            $this->count = $failedValidator->count;
+            $this->value = $failedValidator->value;
+            $this->reason = $failedValidator->reason;
+            $this->records = $failedValidator->records;
             return false;
         }
 
-        $validator = new self($this->target, Record::TYPE_CAA, $this->server);
-        return $validator->isValid($parent);
+        return true;
     }
 
-    private function extractCAAValue(string $rdata): string
+    /**
+     * Validate with a specific DNS server
+     *
+     * @param mixed $value
+     * @param string $dnsServer
+     * @return bool
+     */
+    protected function isValidWithDNSServer(mixed $value, string $dnsServer): bool
```

Comment:
Same as before, seems like helper method only used once. We dont need to abstract here.